tests/e2e/test_misc_commands.py

```diff
@@ -76,20 +81,19 @@ def test_protostar_version_in_config_file(mocker, protostar_bin: Path):
         assert version_manager.protostar_version == protostar_version
 
 
-@pytest.mark.usefixtures("init")
-@pytest.mark.parametrize("protostar_version", ["0.3.0"])
-@pytest.mark.parametrize("protostar_toml_protostar_version", ["0.2.8"])
-@pytest.mark.parametrize("latest_supported_protostar_toml_version", ["0.2.9"])
-@pytest.mark.parametrize("command", ["build", "install", "test"])
-def test_protostar_asserts_version_compatibility(protostar, command):
-    output = protostar([command], expect_exit_code=1)
-    assert "is not compatible with provided protostar.toml" in str(output)
+def test_protostar_version_in_correct_format(protostar):
+    res = protostar(["--v"])
 
+    match = re.match(r".*Cairo-lang version: (.*?)\n.*", res, flags=re.DOTALL)
+    assert match, "Cairo-lang version string not found"
+    assert (
+        len(match.groups()) == 1
+    ), f"There should be only one group, found: {match.groups()}"
 
-@pytest.mark.usefixtures("init")
-@pytest.mark.parametrize("protostar_version", ["0.4.0"])
-@pytest.mark.parametrize("protostar_toml_protostar_version", ["0.3.0"])
-@pytest.mark.parametrize("latest_supported_protostar_toml_version", ["0.3.0"])
-@pytest.mark.parametrize("command", ["build", "install", "test"])
-def test_protostar_passes_version_check_on_compatible_v(protostar, command):
-    protostar([command])
+    v_string = match.group(1)
+    v_object = parse_version(v_string)
+
+    # The v_object is LegacyVersion instead, when regex matching fails (as does in case of ^0.10.0)
+    assert isinstance(
+        v_object, Version
+    ), f"Output version ({v_string}) does not meet the format requirements"
```

Comment:
Nit: Version format could've been checked by regex without using parse_version.

Reply:
That's true but parse_version's regex is super elaborate - check the sources

Reply:
It includes all of the version's weirdnesses